src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/TypeNameCheck.java

```diff
@@ -57,6 +62,7 @@
  * ENUM_DEF</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
  * ANNOTATION_DEF</a>.
+ * Type - {@code TokenTypes}
```

Comment:
Please add to this PR, some Check that has custom type of property as enum.

Type should be fully qualified class name.

`TokenTypes`   should be `subsetOf TokenTypes` to make is clear that type is very very specific, it is not only set of tokens, but not all values are allowed.

Reply:
Done